contracts/EverlongStrategy.sol

```diff
@@ -142,17 +142,38 @@ contract EverlongStrategy is BaseStrategy {
     }
 
     /// @dev Attempt to free the '_amount' of 'asset'.
-    /// @dev Any difference between `_amount` and what is actually freed will be
-    ///      counted as a loss and passed on to the withdrawer.
+    ///      - Any difference between `_amount` and what is actually freed will be
+    ///        counted as a loss and passed on to the withdrawer.
+    ///      - Unrealized losses must be calculated and proportionally
+    ///        distributed. Otherwise early withdrawers will incur no losses,
+    ///        and the last to withdraw will incur all losses.
     /// @param _amount The amount of 'asset' to be freed.
     function _freeFunds(uint256 _amount) internal override {
-        // Close all matured positions (if any).
-        uint256 output = _closeMaturedPositions(0);
-
-        // Close immature positions if additional funds need to be freed.
-        if (_amount > output) {
-            _closePositions(_amount - output);
+        // Calculate the current `totalAssets` and retrieve the previous value.
+        uint256 idle = asset.balanceOf(address(this));
+        uint256 currentTotalAssets = calculatePortfolioValue() + idle;
+        uint256 previousTotalAssets = TokenizedStrategy.totalAssets();
+
+        // If the current `totalAssets` is less than the previous, there are
+        // unrealized losses.
+        if (currentTotalAssets < previousTotalAssets) {
+            // Calculate the withdrawer's proportion of losses.
+            //
+            // It's important to use the TOTAL withdrawal amount, not just the
+            // amount being freed, when calculating the withdrawer's share.
+            //
+            //     totalWithdrawalAmount = _amount + idle
+            //
+            uint256 loss = previousTotalAssets - currentTotalAssets;
+            uint256 proportionalLoss = (loss).mulDivDown(
+                _amount + idle,
```

Comment:
Just to make sure I understand this logic, I'll summarize my understanding of this calculation.

The gist is simple ofc, we just want to find the loss that is appropriate for this trader to pay. We just scale the total loss to the portion of funds that are actually being moved. My questions have to do with why we use `_amount + idle` and `previousTotalAssets` to calculate this proportion. 

`_amount + idle` is the amount of funds being withdrawn (`_freeFunds` is called during a withdrawal, and we know that idle is prioritized). This makes sense. Next up is `previousTotalAssets`. This was initially confusing to me, but it makes sense considering that `totalAssets` is used to compute the vault share price. 

IMO it would be good to make a comment justifying why we use `previousTotalAssets`

Reply:
Yeah so this one definitely put my head in a blender a bit....

The redeemer's proportional share of the portfolio losses is as follows (assuming losses have occurred) :
```
redeemer_loss_share = total_portfolio_losses * value_of_longs_closed / (previous_portfolio_value - current_portfolio_value)
```

Unfortunately `totalAssets` is a combination of portfolio value and idle assets, and we don't know the proportion of each when looking at the stored value.... hugely problematic if idle liquidity varied dramatically, but it doesn't.

For our case, the strategy will almost always have zero idle (except right after a redemption due to the `partialPositionClosureBuffer`), and the maximum amount of idle it can have is just under Hyperdrive's `minimumTransactionAmount`. 

Since idle liquidity won't vary greatly and has little effect on `totalAssets`, I think it's safe for us to just compare `totalAssets` to determine bond portfolio losses.